### PR TITLE
Add optional Route53 CNAME for status page

### PR DESCRIPTION
**Pull request by mmarcotte**
This introduces a new environment variable `STATUSPAGE_DNS_RECORD`. If it is set, Terraform will attempt to create a Route53 CNAME record with its value in order to direct users for `status.${EFCMS_DOMAIN}` (e.g., https://support.dawson.ustaxcourt.gov). 

This way if AWS experiences an outage, EFCMS experiences an outage for any reason, we can have this page live in order to communicate system status. 

Aiming to satisfy #305

### CODE REVIEW

File: web-client/terraform/main/main.tf [hunk 1]
```diff
@@ -122,3 +122,14 @@ resource "aws_acm_certificate_validation" "dns_validation_west" {
   validation_record_fqdns = [aws_route53_record.record_certs.fqdn]
   provider                = aws.us-west-1
 }
+
+resource "aws_route53_record" "statuspage" {
+  count   = var.statuspage_dns_record != "" ? 1 : 0
```
Comment (julialeague) [inline web-client/terraform/main/main.tf:127]: neat use of count!

File: web-client/terraform/main/variables.tf [hunk 2]
```diff
@@ -40,4 +40,9 @@ variable "is_dynamsoft_enabled" {
 
 variable "zone_name" {
   type = string
+}
```
Comment (julialeague) [inline web-client/terraform/main/variables.tf:43]: nit: new line at the end of the files?

**Comment by mmarcotte** [comment]
from a terraform plan:
```
# aws_route53_record.statuspage[0] will be created
  + resource "aws_route53_record" "statuspage" {
      + allow_overwrite = (known after apply)
      + fqdn            = (known after apply)
      + id              = (known after apply)
      + name            = "status.dawson.ustaxcourt.gov"
      + records         = [
          + "***",
        ]
      + ttl             = 60
      + type            = "CNAME"
      + zone_id         = "***"
    }
```

File: web-client/terraform/main/main.tf (hunk 1 above)
Comment (mmarcotte) [reply web-client/terraform/main/main.tf:127]: yea i was trying to figure out how to make this optional. looks like this worked as expected. pretty cool. thanks for the nudge in the right direction @adunkman

File: web-client/terraform/main/main.tf [hunk 3]
```diff
@@ -122,3 +122,14 @@ resource "aws_acm_certificate_validation" "dns_validation_west" {
...
+  name    = "status.${var.dns_domain}"
+  type    = "CNAME"
+  zone_id = data.aws_route53_zone.zone.zone_id
+  ttl     = 60
+  records = [
+    var.statuspage_dns_record
+  ]
+}
```
Comment (julialeague) [inline web-client/terraform/main/main.tf]: new line here too if you don't mind

**Review by julialeague — APPROVED** [review]

**Pull request merged**